token-lending/program/src/processor.rs

```diff
@@ -1338,8 +1340,49 @@ fn _withdraw_obligation_collateral<'a>(
         return Err(LendingError::InvalidMarketAuthority.into());
     }
 
+    // account for lending market and reserve rate limiter when withdrawing. this is needed to
+    // support max withdraws.
+    let max_outflow_collateral_amount = if account_for_rate_limiter {
+        let max_outflow_usd = lending_market
+            .rate_limiter
+            .clone() // remaining_outflow is a mutable call, but we don't have mutable access here
+            .remaining_outflow(clock.slot)?;
+
+        let max_lending_market_outflow_liquidity_amount = withdraw_reserve
+            .usd_to_liquidity_amount_lower_bound(min(
+                max_outflow_usd,
+                // min here bc this function can overflow if max_outflow_usd is u64::MAX
+                // the actual value doesn't matter too much as long as its sensible
+                obligation.deposited_value.try_mul(2)?,
```

Comment:
multiplied by 2 because multiplying by 1 caused some annoying off-by-one token ux when withdrawing which i didn't like

Reply:
oh idt that actually works, becuase u64max - 1 could also overflow. gonna justt leave it as is